addons/mrp/models/mrp_workorder.py

```diff
@@ -437,7 +438,7 @@ def write(self, values):
                 if workorder.workcenter_id.id != values['workcenter_id']:
                     if workorder.state in ('progress', 'done', 'cancel'):
                         raise UserError(_('You cannot change the workcenter of a work order that is in progress or done.'))
-                    workorder.leave_id.resource_id = self.env['mrp.workcenter'].browse(values['workcenter_id']).resource_id
+                    workorder.leave_id.resource_id = self.env['mrp.workcenter'].browse(values['workcenter_id']).resource_ids[0] # not sure here if we need to take the first one or if we need to compute the most optimized one?
```

Comment:
Would be better to find if a ressource has available slot but could be in another task